lib/pubsubstub/stream.rb

```diff
@@ -0,0 +1,26 @@
+class Stream
+  def initialize(&callback)
+    @callback = callback
+    @closed = false
+  end
+
+  def close
+    return if closed?
+    @closed = true
+  end
+
+  def each(&front)
+    @front = front
+    @callback.call(self)
+    close unless @keep_open
```

Comment:
Is the closing part taken of by itself when trying to write to a disconnected socket, raising an exception and unregistering the subscription? If yes, is `close` even necessary?